### PR TITLE
Allow attaching to profile builds

### LINKED ISSUE #26711: flutter attach to profile build should work
Currently trying to use flutter attach on a running profile build results in an exception when the DevFS cannot be setup:

```
Error -32603 received from application: Internal error
Error initializing DevFS: JSON-RPC error -32603 (internal error): Internal error
```

At the very least we should throw a better exception here.

### PR DESCRIPTION
**Pull request by jonahwilliams**
This is required for profiling embedded flutter views. Fixes https://github.com/flutter/flutter/issues/26711 and internal issue b/122984037

### CODE REVIEW

File: packages/flutter_tools/lib/src/run_cold.dart [hunk 1]
```diff
@@ -111,12 +113,50 @@ class ColdRunner extends ResidentRunner {
...
+  Future<int> attach({
+    Completer<DebugConnectionInfo> connectionInfoCompleter,
+    Completer<void> appStartedCompleter,
+  }) async {
+    _didAttach = true;
+    try {
+      await connectToServiceProtocol();
+    } catch (error) {
+      printError('Error connecting to the service protocol: $error');
+      return 2;
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/run_cold.dart]: what does returning 2 mean?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/run_cold.dart]: I believe it is a signal used by the daemon. I will find the source and give this and the version in run_hot a doc comment

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/run_cold.dart]: I copied it from run_hot attach, maybe you know from https://github.com/flutter/flutter/pull/8764 ? It doesn't seem to go anywhere

Comment (zanderso) [reply packages/flutter_tools/lib/src/run_cold.dart]: I probably copied it from somewhere as well back then, unfortunately.

**Review by zanderso — APPROVED** [review]

**Pull request merged**